Bluejay.asm

```diff
@@ -1743,191 +1760,213 @@ scheduler_128ms_switch_case_32ms:
+	jc	scheduler_steps_even_telemetry_demag_metric	; Yes - Jump to next scheduler
 
 	mov	Temp_Pwm_Level_Setpoint, #100				; No - update pwm limit (about 40% allowing landing)
 
 	subb	A, #(TEMP_LIMIT_STEP / 2)				; Is temperature below final limit
-	jc	scheduler_1024ms_switch_case				; Yes - Jump to next scheduler
+	jc	scheduler_steps_even_telemetry_demag_metric	; Yes - Jump to next scheduler
 
 	mov	Temp_Pwm_Level_Setpoint, #50				; No - update pwm limit (about 20% forced landing)
 	; Zero pwm cannot be set because of set_pwm_limit algo restrictions
 	; Otherwise hard stuttering is produced
 
-	; Jump to next scheduler
-	sjmp scheduler_1024ms_switch_case
+scheduler_steps_even_telemetry_demag_metric:
+	; ********************* [TELEMETRY] SEND DEMAG METRIC FRAME *****************
+	mov Ext_Telemetry_L, Demag_Detected_Metric		; Set telemetry low value to demag metric data
+	mov Ext_Telemetry_H, #0Ch						; Set telemetry high value to demag metric frame ID
+
+	; No more work to do
+	jmp scheduler_exit
 
-scheduler_128ms_switch_case_64ms:
-	cjne A, #2, scheduler_128ms_switch_case_96ms
+scheduler_steps_odd:
+	; ********************* UPDATE TEMPERATURE PWM LIMIT *****************
 
 	; Check temp protection enabled, and exit when protection is disabled
 	mov	A, Temp_Prot_Limit
-	jz	scheduler_1024ms_switch_case
-
-	; Update PWM limit
+	jz	scheduler_steps_odd_choose_step
 
 	; pwm limit is updated one unit at a time to avoid abrupt pwm changes
-	; resulting in current spikes
+	; resulting in current spikes, that may damage motor/esc
 	; Compare pwm limit to setpoint
 	clr C
 	mov	A, Pwm_Limit
 	subb A, Temp_Pwm_Level_Setpoint
-	jz scheduler_1024ms_switch_case					; pwm limit == setpoint -> next
-	jc scheduler_128ms_temp_update_pwm_limit_inc	; pwm limit < setpoint -> increase pwm limit
+	jz scheduler_steps_odd_choose_step				; pwm limit == setpoint -> next
+	jc scheduler_steps_odd_temp_pwm_limit_inc		; pwm limit < setpoint -> increase pwm limit
 
-scheduler_128ms_temp_update_pwm_limit_dec:
+scheduler_steps_odd_temp_pwm_limit_dec:
 	; Decrease pwm limit
-	mov A, Pwm_Limit
-	jz scheduler_1024ms_switch_case					; pwm limit is 0 -> Exit
 	dec Pwm_Limit
 
-	; Jump to next scheduler
-	sjmp scheduler_1024ms_switch_case
+	; Now run speciffic odd scheduler step
+	sjmp scheduler_steps_odd_choose_step
 
-scheduler_128ms_temp_update_pwm_limit_inc:
+scheduler_steps_odd_temp_pwm_limit_inc:
 	; Increase pwm limit
-	mov A, Pwm_Limit
-	inc A
-	jz ($+4)
 	inc Pwm_Limit
 
-	; Jump to next scheduler
-	sjmp scheduler_1024ms_switch_case
+	; Now run speciffic odd scheduler step
 
-scheduler_128ms_switch_case_96ms:
-	cjne A, #3, scheduler_1024ms_switch_case
-
-	; Start a new ADC conversion so after 64ms it will be ready for stage 2 og 128ms scheduler
-	Stop_Adc
-	Start_Adc
-
-	; Continue on next scheduler
+scheduler_steps_odd_choose_step:
+	; Let A = Scheduler_Counter % 8, so A = [0 - 7] value
+	mov A, Scheduler_Counter
+	anl A, #07h
 
+scheduler_steps_odd_step_1:
```

Comment:
I am not super happy with the labels. I know I already complained about odd/even, but maybe we could rename the labels to what's actually happening?

Maybe instead of `scheduler_steps_odd_step_1` we could name it `scheduler_steps_odd_status_frame`?

Reply:
I agree, I have changed them for others following the naming proposed